src/mds/MDCache.cc

```diff
@@ -9812,6 +9812,19 @@ void MDCache::dispatch_request(const MDRequestRef& mdr)
 {
   if (mdr->dead) {
     dout(20) << __func__ << ": dead " << *mdr << dendl;
+    //if the mdr is a "batch_op" and it has followers, pick a follower as
+    //the new "head of the batch ops" and go on processing the new one.
+    if (mdr->client_request && mdr->is_batch_head()) {
+      int mask = mdr->client_request->head.args.getattr.mask;
+      auto it = mdr->batch_op_map->find(mask);
+      auto new_batch_head = it->second->find_new_head();
+      if (!new_batch_head) {
+	mdr->batch_op_map->erase(it);
+	return;
+      }
```

Comment:
dout(..) here - might help during debug.

Reply:
Makes sense and will add it.